lib/api/environment.rb

```diff
@@ -19,12 +19,12 @@ def self.time_attributes
         next if cspec[:klass].blank?
         klass = nil
 
-        # Temporary measure to avoid thread race condition which could lead to a deadlock
-        ActiveSupport::Dependencies.interlock.permit_concurrent_loads do
+        # Ensure we're the only thread trying to autoload classes and their columns
+        ActiveSupport::Dependencies.interlock.loading do
```

Comment:
Note, this also works:

```ruby
        Rails.application.executor.wrap do
...
```

Reply:
@Fryguy do you prefer the executor wrapper?  I can research more tomorrow if that makes sense over the interlock.loading.

Reply:
Yes, the executor wrapper seemed like a better approach.

Reply:
haha, executor.wrap deadlocks if I hit compute -> cloud after login  🤣 ... back to `.loading`